google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/RetryingConnectionImpl.java

```diff
@@ -171,6 +178,14 @@ public final void onError(Throwable t) {
       setPermanentError(statusOr.get());
       return;
     }
+    if (ResetSignal.isResetSignal(statusOr.get())) {
```

Comment:
Can you instead modify the triggerReinitialize method to pass the previous failure status? Then you can isolate this logic to the subscriber which needs to handle it.

Reply:
Done